### DISCUSSION TITLE
Question about using useFieldArray and useForm

### DISCUSSION THREAD

**Discussion by marianozicavo** [react-hook-form/react-hook-form#4848] [Q&A] [answered]
I'm setting a default array value using `useForm` and then I'm using `useFieldArray` to manage that array.

```js
  const { control, watch, handleSubmit } = useForm({
    defaultValues: {
      test: [{ firstName: "firstName", lastName: "lastName" }]
    }
  });
  const { fields } = useFieldArray({
    control: control,
    name: "test"
  });
  ```
On the first render, this is the result of `console.log(watch('test'), fields)`
![image](https://user-images.githubusercontent.com/7256996/115135608-22d0aa80-9ff0-11eb-8c25-03188d023fc9.png)
Both have the default value.

But a rerender is being immediately triggered and now `console.log(watch('test'), fields)` logs the following:
![image](https://user-images.githubusercontent.com/7256996/115135675-878c0500-9ff0-11eb-8405-4f55d5a6e97d.png)

Even `handleSubmit` is receiving `test` with an empty array.

https://codesandbox.io/s/simple-mmtbq?file=/src/index.js

Am I using `useFieldArray` correctly? Should I access `fields` directly from the `handleSubmit` function in order to retrieve those values?

Thanks in advance!

**Comment by bluebill1049** [comment, marked as answer]
because there are no fields been registered in hook form, watch is return defaultValues in the first render and then return what's in the form, take a look the tip/rules at watch section in the doc.

**Reply by marianozicavo** [reply]
OK I see.
My problem was that I was using `useFieldArray` in the content of a dialog. It looks like Material UI's Dialog doesn't actually render the Dialog content on the first render, even if the dialog is hardcoded to be open. So I was calling the hook without actually rendering the inputs.

Thanks for the answer!
